app/Http/Controllers/User/ArticleController.php

```diff
@@ -101,7 +105,19 @@ public function show($id)
      */
     public function edit($id)
     {
-        //
+        $article = $this->articleRepository->getArticle($id);
+        if (Gate::allows('retrieve-article', $article)) {
+            $categories = $this->categoryRepository->getCategoryList();
+            $articleCategories = $article->categories;
+
+            return view('user.article.edit', compact(
+                'article',
+                'categories',
+                'articleCategories'
+            ));
+        } else {
+            return back()->with('error', __("You're not allow to retrieve this article"));
+        }
```

Comment:
```suggestion
        }
        
        return back()->with('error', __("You're not allow to retrieve this article"));
```
fix hết mấy chỗ tương tự nhé